.github/workflows/go.yml

```diff
@@ -0,0 +1,28 @@
+# This workflow will build a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: Go
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.20'
```

Comment:
Will it be good to use `1.22` here? since we are using go 1.22 in the go mod file [here](https://github.com/goprojs/product_catalog/blob/8422275a0f281e2ed4e39467580bd582300bd9d2/backend_api/go.mod#L3C1-L3C10)